pkg/util/test/util.go

```diff
@@ -288,6 +288,42 @@ func ValidateStorageCluster(
 
 	pxEndpoint = fmt.Sprintf("%s:%d", svc.Spec.ClusterIP, servicePort)
 
+	if err := validatePodsByName(cluster.Namespace, "portworx-pvc-controller", timeout, interval); err != nil {
+		return err
+	}
+
+	if cluster.Spec.Stork != nil && cluster.Spec.Stork.Enabled {
+		if err := validatePodsByName(cluster.Namespace, "stork", timeout, interval); err != nil {
+			return err
+		}
+		if err := validatePodsByName(cluster.Namespace, "stork-scheduler", timeout, interval); err != nil {
+			return err
+		}
+	}
+
+	if cluster.Spec.Autopilot != nil && cluster.Spec.Autopilot.Enabled {
+		if err := validatePodsByName(cluster.Namespace, "autopilot", timeout, interval); err != nil {
+			return err
+		}
+	}
+
+	if cluster.Spec.UserInterface != nil && cluster.Spec.UserInterface.Enabled {
+		if err := validatePodsByName(cluster.Namespace, "lighthouse", timeout, interval); err != nil {
+			return err
+		}
+	}
+
+	if csi, ok := cluster.Spec.FeatureGates["CSI"]; ok && csi == "true" {
+		if err := validatePodsByApp(cluster.Namespace, "px-csi-driver", timeout, interval); err != nil {
```

Comment:
We should also probably validate `portworx` pods here, as they will have extra containers running if `CSI` is enabled.